elastic/kibana
# [ResponseOps][Alerts] Add null-value bucket detection to alerts group aggregations endpoint

## Summary

- Adds null-value bucket detection to server-side alerts aggregations and marks those groups with a `--` key and `isNullGroup = true`.
- Improves alerts grouping types with default aggregations.
- Improves documentation

## To verify

1. Temporarily merge [#189958](https://github.com/elastic/kibana/pull/189958) into this branch
2. Create a rule that fires alerts in Observability > Alerts (i.e. Custom Threshold, ES Query, ...)
3. Once you start to see some alerts in the Alerts page, toggle the grouped alerts view using the dropdown at the top-right of the table (`Group alerts by: ...`), selecting a custom field that doesn't have a value in alert documents (to find one, open the alert flyout and look at the fields table)
4. Check that the group based on the empty field shows `--` as a title
5. Check that the alerts table in the expanded group panel is filtered correctly

### References

Refs [#189958](https://github.com/elastic/kibana/pull/189958)

### Checklist

Delete any items that are not applicable to this PR.
